src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -395,12 +381,12 @@ internal sealed class MemberSignatureComparer : IEqualityComparer<Symbol>
             _considerReturnType = considerReturnType;
             _considerTypeConstraints = considerTypeConstraints;
             _considerCallingConvention = considerCallingConvention;
-            _considerRefKindDifferences = considerRefKindDifferences;
+            _refKindCompareMode = refKindCompareMode;
             _considerArity = considerArity;
             _typeComparison = typeComparison;
             Debug.Assert((_typeComparison & TypeCompareKind.FunctionPointerRefMatchesOutInRefReadonly) == 0,
-                         $"Rely on the {nameof(considerRefKindDifferences)} flag to set this to ensure all cases are handled.");
-            if (!considerRefKindDifferences)
+                         $"Rely on the {nameof(refKindCompareMode)} flag to set this to ensure all cases are handled.");
+            if ((refKindCompareMode & RefKindCompareMode.ConsiderDifferences) == 0)
```

Comment:
Could we assert that valid combinations of flags are used? (`AllowRefReadonlyVsInMismatch` can't be set without `ConsiderDifferences`)